.mergify.yml

```diff
@@ -1,15 +1,55 @@
 pull_request_rules:
   - name: Automatic merge on approval
     conditions:
-      - "#approved-reviews-by>=2"
-      - check-success=Validation
-      - check-success=Tests Normal
-      - check-success=Tests Madsim
-      - check-success=Commit Message Validation
-      - check-success=Spell Check
-      - check-success=DCO
-      - status-success="codecov/patch"
-      - status-success="codecov/project"
+      - or:
+        - "#approved-reviews-by >= 2"
+        - and:
+          - author = dependabot[bot]
+          - "#approved-reviews-by>=1"
+      - "#check-failure=0"
     actions:
       merge:
         method: rebase
+
+  - name: convert to draft
+    conditions:
+      - and:
+          - "#check-failure>0"
+          - -check-failure~=codecov/.*
+    actions:
+      comment:
+        message: "@{{author}} Convert your pr to draft since CI failed"
+      edit:
+        draft: true
+
+  - name: automatic approval for Dependabot pull requests
```

Comment:
Will this cause failed depbot PR be merged automatically?

Reply:
It won't since a failed depbot PR doesn't  meet the "Automatic merge on approval" rule